### PR TITLE
feat: brake fan

### PR DESCRIPTION
**Pull request by tyler58546**
Fixes #[issue_no]

## Summary of Changes

- [x] Enable brake fan button
- [x] Update ECAM messages
- [x] Animate brake fan button
- [x] Implement brake cooling
- [x] Add HOT label to button
- [x] Improve landing gear textures

Sounds will be added in a separate PR from Boris

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Test to make sure the following conditions are met:

- If the switch is engaged with the gear up, it will remain engaged physically but the brake fans will not start, the brake fans will start automatically if the gear is down and locked (more specifically the left main landing gear).
- If the switch is engaged and the gear is down, it will remain engaged physically and the brake fans will run, however if the gear is then retracted, the brake fans should stop running.
- If the switch is disengaged the brake fans shouldn't run under any condition.
- The brake fan switch will display "HOT" if at least one of the brakes is above 300°C.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Inexorable95** [comment]
The brake cooling mechanics have been added (**this only concerns brake cooling, not brakes heat gain**)
**For testing purposes the temps are displayed as they are on the wheel ECAM page and not rounded to the nearest five (100, 105, 110, etc)**
**Also, real brake temps are displayed for test purposes, this will ultimately be reverted for release.**
Brakes cool according to :  

- ambiant temp (the further they are for ambiant temp, the faster they will cool)
- airspeed (higher winds or increased airplane speed will make the brake cool faster)
- whether or not the brake fan is on.

**Special note for brake fan use:** 
**From airbus :**
_the temperature indicated in the cockpit when the brake fans are running is significantly lower than the temperature indicated when the brake fans are OFF. This is because the brake fans are ventilating the brake temperature sensor that is not located inside the material itself. Therefore, as soon as the brake fans are running, the indicated brake temperature decreases almost instantaneously (except ABS). Similarly, it will take several minutes for the indicated temperature to rise, and to match the actual brake temperature, when brake fans are switched OFF.
For all brake types (except ABS), when the fans are running, the difference ranges typically from approximately 50°C at 100°C actual brake temperature up to about 150°C at 300°C actual brake temperature. As a consequence, if the BRAKE HOT warning is activated just before takeoff, the ECAM will request the selection of the brake fans, if installed, and the delay of the takeoff. When fans are selected, the warning will disappear instantaneously, but the takeoff must be delayed until the indicated temperature consistently decreases below 150°C (for a warning set at 300°C). Then, the brake fans should be switched OFF for takeoff. This is well documented in the FCOM._

Because of this, the following may occur:
- When the fans are on, the reported temperature (probe) will fall faster than the real brake temperature
- When the fans are switched off, the reported brake temperature (probe) will rise up even though no brake pressure is applied (that is the probe equalizing with the brake temperature)

**Observations expected :**
With negligable wind, an ambiant temp around 15°C, the brakes should **approximately** follow this official chart (the temperatures are in fahrenheit):
brake_curves_original

For reference, the values of the chart are as follow: 

FAN OFF :
0m: 1070°F / 577°C
10m: 840°F / 450°C
20m: 690°F / 365°C
30m: 595°F / 312°C
40m: 535°F / 280°C
50m: 490°F / 254°C
60m: 430°F / 221°C

FAN ON: 
0m: 1070°F / 577°C
10m: 685°F / 363°C
20m: 495°F / 257°C
30m: 395°F / 201°C
40m: 310°F / 154°C
50m: 255°F / 123°C
60m: 195°F / 91 °C

In case of hot brakes during flight, flying with the gear extended and the brake fans off at 250kts and FL180 should be approximately as fast as cooling with the fan on on the ground (see the example above)(we don't have a lot of data about this, mainly pilots experience collected in the #pilot-feedback channel).

Please comment if you find any mistake in this, it has been validated by pilots from the discord but we never know :)

**Review by pepperoni505 — CHANGES_REQUESTED** [review]
You undid a few changes made to the decals, if possible can you reapply the changes you made to the latest decal files?

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (pepperoni505) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:996]:
```suggestion
                    <TOOLTIPID>%((L:A32NX_BRAKE_FAN_BTN_PRESSED, Bool))%{if}Turn ON brake fan%{else}Turn OFF brake fan%{end}</TOOLTIPID>
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html
Comment (pepperoni505) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html:264]: You should remove this

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (Inexorable95) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:996]: The brake fan isn't 100% tied to the button, for instance if the gear is up, the brake fan will not be running even if the button is pressed. So I think the text "Engage/release the brake fan button" is closer to what the button actually does

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html
Comment (Inexorable95) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html:264]: We agreed with tyler that we should keep this for the QA testers, just to make sure everything works correctly regarding the brake cooling simulation. It will be removed before merging

**Comment by Inexorable95** [comment]
> You undid a few changes made to the decals, if possible can you reapply the changes you made to the latest decal files?
> 
> There also seems to be a decal issue:
> image

Should be fixed

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (pepperoni505) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:996]: I'm not really a fan of the current wording, is there anything else you can think of?

Also, you can't engage auto brake max in the air but it still says "Set auto-brake to MAX"

File: src/model/a320-interior/A320_NEO_INTERIOR_LOD00.gltf
Comment (pepperoni505) [inline src/model/a320-interior/A320_NEO_INTERIOR_LOD00.gltf:157665]: This will cause issues, as meshes are referenced by index

**Comment by pepperoni505** [comment]
I'm marking this as a draft until we resolve the glTF issues

**Review by pepperoni505 — APPROVED** [review]
LGTM

**Comment by pepperoni505** [comment]
Just a note for once this is finished testing, let's hold back until after World Update 3 to merge this as Asobo may animate their brake fan button

**Comment by GlitchedYT** [comment]
Category: QA Tester
Name: Glitched#0740
Date of testing: Current date 02/15/2021
Version of the sim: 1.12.13.0
PR Tested: #3176
Tier of testing: 2
Changes to observe:  
 Enable brake fan button
 Update ECAM messages
 Animate brake fan button
 Implement brake cooling
 Add HOT label to button
 Improve landing gear textures
(Insert the Summary of changes, in order to remind what is to test)

Testing technique: If the switch is engaged with the gear up, it will remain engaged physically but the brake fans will not start, the brake fans will start automatically if the gear is down and locked (more specifically the left main landing gear).
If the switch is engaged and the gear is down, it will remain engaged physically and the brake fans will run, however if the gear is then retracted, the brake fans should stop running.
If the switch is disengaged the brake fans shouldn't run under any condition.
The brake fan switch will display "HOT" if at least one of the brakes is above 300°C.

Steps you've made to test the Pull Request:
1. Spawned on the runway at KLAX
2. Did two rejected takeoff's to make the brakes heat up
3. As brake's went above 300°C "HOT" Appeared on brake fan button
4. Took off, put gear up, turned on brake fan and noticed brake temps didn't change
5. Lowered gear, when Left Maine Gear locked brake button turned on and started cooling brakes

Flight notes (When done a full flight):

Media: Insert screenshots, recordings, or anything relevant to the review here.
Microsoft Flight Simulator Screenshot 2021 02 15 - 16 35 17 99
Microsoft Flight Simulator Screenshot 2021 02 15 - 16 35 39 63
Microsoft Flight Simulator Screenshot 2021 02 15 - 16 36 09 88


Issues (If Any): Not part of the PR but noticed Max Autobrake doesn't do anything during a rejected T/O
Steps to achieve the issue
Perform a rejected takeoff with AUTO BRK MAX armed

Overall Rating: Good, we finally have a break fan:) thank you very much Tyler!!
P.S. I still kind of think the actual tire textures could be improved further.. but other than that the new brake textures look great!
Conclusions of the testing: Make a small descriptive to areas where it needs to be improved and recommendations.

**Comment by Inexorable95** [comment]
LGTM

**Comment by Inexorable95** [comment]
LGTM

**Comment by cianwhite** [comment]
Category: QA Tester
Name: Cian#3596
Date of testing: 15/02/2021
Version of the sim: 1.12.13.0
PR Tested: #3176 
Tier of testing: 2
Changes to observe: Brake fan implementation 

Testing technique: Started C&D and performed circuits observing brake fans.
Steps you've made to test the Pull Request:
1. Started C&D at EIDW, performed a RTO and as stated above MAX AB isn't working.
2. Took off an performed circuits, the brake fan logic as outlined works as it should.
3. Landed and watched brakes go above 300 degrees, triggering a HOT BRK fan button. 
4. Observed cooling which works as the table suggests.

Flight notes (When done a full flight): N/A

Media: 
Screenshot_13
Ref: 
Screenshot_14

Issues (If Any): Brake fan should have an ON outline similar to the autobrake. Also as outlined above max autobrake is INOP, medium and low seemed to work fine however.
Steps to achieve the issue: Look at brake fan button when it is not on.

Overall Rating: To be improved
Conclusions of the testing: The brake fan logic works as outlined, however the modelling on the button seem to be wrong as stated above (I'm not sure if this is possible as of right now to properly implement).

**Comment by GlitchedYT** [comment]
As another PSA AUTO BRK MAX not working is not due to this PR, is not working in custom ap branch either. Refer to the bug report i have created here: https://github.com/flybywiresim/a32nx/issues/3546

**Comment by Inexorable95** [comment]
The changelog has been updated and the brake fan now has the grey "on" label when the brake fans are off
brakefannew

**Comment by pepperoni505** [comment]
The MSFS update didn't include any glTF file changes, so I am removing the Do Not Merge label

**Review by Inexorable95 — APPROVED** [review]

**Review by Inexorable95 — APPROVED** [review]

**Pull request merged**
